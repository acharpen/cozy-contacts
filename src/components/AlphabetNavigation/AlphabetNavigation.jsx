import React, { useEffect, useState } from 'react'
import cx from 'classnames'
import IconButton from 'cozy-ui/transpiled/react/IconButton'
import { get } from 'lodash'
import PropTypes from 'prop-types'

const AlphabetNavigation = ({ contacts, current, onNavigate }) => {
  const [firstLetters, setFirstLetters] = useState([])

  useEffect(
    () => {
      setFirstLetters(
        contacts
          .map(contact => {
            return get(contact, 'name.familyName', null)
          })
          .filter(familyName => !!familyName)
          .map(familyName => {
            return familyName.charAt(0).toUpperCase()
          })
      )
    },
    [contacts]
  )

  const alphabet = [...Array(26)].map((_, i) => {
    const letter = String.fromCharCode(i + 65)
    return {
      letter,
      isActive: letter === current,
      isDisabled: !firstLetters.includes(letter)
    }
  })

  return (
    <div className="alphabet-navigation">
      {alphabet.map(({ letter, isActive, isDisabled }) => (
        <div
          key={letter}
          className={cx('alphabet-letter', { active: isActive })}
        >
          <IconButton
            color="primary"
            size="small"
            disabled={isDisabled}
            onClick={() => onNavigate(letter === current ? null : letter)}
          >
            {letter}
          </IconButton>
        </div>
      ))}
    </div>
  )
}

AlphabetNavigation.propTypes = {
  contacts: PropTypes.array.isRequired,
  current: PropTypes.string,
  onNavigate: PropTypes.func.isRequired
}

export default AlphabetNavigation
