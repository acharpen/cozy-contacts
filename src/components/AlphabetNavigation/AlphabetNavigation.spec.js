import React from 'react'
import AlphabetNavigation from './AlphabetNavigation'
import renderer from 'react-test-renderer'

it('should match the snapshot', () => {
  const contacts = [{ name: { familyName: 'Doe' } }]

  const tree = renderer
    .create(<AlphabetNavigation contacts={contacts} onNavigate={() => {}} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
