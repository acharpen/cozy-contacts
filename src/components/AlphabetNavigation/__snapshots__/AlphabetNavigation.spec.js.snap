// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<div
  className="alphabet-navigation"
>
  <div
    className="alphabet-letter"
  >
    <button
      className="styles__IconButton___1LWwV"
      color="primary"
      disabled={true}
      onClick={[Function]}
      size="small"
    >
      A
    </button>
  </div>
  <div
    className="alphabet-letter"
  >
    <button
      className="styles__IconButton___1LWwV"
      color="primary"
      disabled={true}
      onClick={[Function]}
      size="small"
    >
      B
    </button>
  </div>
  <div
    className="alphabet-letter"
  >
    <button
      className="styles__IconButton___1LWwV"
      color="primary"
      disabled={true}
      onClick={[Function]}
      size="small"
    >
      C
    </button>
  </div>
  <div
    className="alphabet-letter"
  >
    <button
      className="styles__IconButton___1LWwV"
      color="primary"
      disabled={true}
      onClick={[Function]}
      size="small"
    >
      D
    </button>
  </div>
  <div
    className="alphabet-letter"
  >
    <button
      className="styles__IconButton___1LWwV"
      color="primary"
      disabled={true}
      onClick={[Function]}
      size="small"
    >
      E
    </button>
  </div>
  <div
    className="alphabet-letter"
  >
    <button
      className="styles__IconButton___1LWwV"
      color="primary"
      disabled={true}
      onClick={[Function]}
      size="small"
    >
      F
    </button>
  </div>
  <div
    className="alphabet-letter"
  >
    <button
      className="styles__IconButton___1LWwV"
      color="primary"
      disabled={true}
      onClick={[Function]}
      size="small"
    >
      G
    </button>
  </div>
  <div
    className="alphabet-letter"
  >
    <button
      className="styles__IconButton___1LWwV"
      color="primary"
      disabled={true}
      onClick={[Function]}
      size="small"
    >
      H
    </button>
  </div>
  <div
    className="alphabet-letter"
  >
    <button
      className="styles__IconButton___1LWwV"
      color="primary"
      disabled={true}
      onClick={[Function]}
      size="small"
    >
      I
    </button>
  </div>
  <div
    className="alphabet-letter"
  >
    <button
      className="styles__IconButton___1LWwV"
      color="primary"
      disabled={true}
      onClick={[Function]}
      size="small"
    >
      J
    </button>
  </div>
  <div
    className="alphabet-letter"
  >
    <button
      className="styles__IconButton___1LWwV"
      color="primary"
      disabled={true}
      onClick={[Function]}
      size="small"
    >
      K
    </button>
  </div>
  <div
    className="alphabet-letter"
  >
    <button
      className="styles__IconButton___1LWwV"
      color="primary"
      disabled={true}
      onClick={[Function]}
      size="small"
    >
      L
    </button>
  </div>
  <div
    className="alphabet-letter"
  >
    <button
      className="styles__IconButton___1LWwV"
      color="primary"
      disabled={true}
      onClick={[Function]}
      size="small"
    >
      M
    </button>
  </div>
  <div
    className="alphabet-letter"
  >
    <button
      className="styles__IconButton___1LWwV"
      color="primary"
      disabled={true}
      onClick={[Function]}
      size="small"
    >
      N
    </button>
  </div>
  <div
    className="alphabet-letter"
  >
    <button
      className="styles__IconButton___1LWwV"
      color="primary"
      disabled={true}
      onClick={[Function]}
      size="small"
    >
      O
    </button>
  </div>
  <div
    className="alphabet-letter"
  >
    <button
      className="styles__IconButton___1LWwV"
      color="primary"
      disabled={true}
      onClick={[Function]}
      size="small"
    >
      P
    </button>
  </div>
  <div
    className="alphabet-letter"
  >
    <button
      className="styles__IconButton___1LWwV"
      color="primary"
      disabled={true}
      onClick={[Function]}
      size="small"
    >
      Q
    </button>
  </div>
  <div
    className="alphabet-letter"
  >
    <button
      className="styles__IconButton___1LWwV"
      color="primary"
      disabled={true}
      onClick={[Function]}
      size="small"
    >
      R
    </button>
  </div>
  <div
    className="alphabet-letter"
  >
    <button
      className="styles__IconButton___1LWwV"
      color="primary"
      disabled={true}
      onClick={[Function]}
      size="small"
    >
      S
    </button>
  </div>
  <div
    className="alphabet-letter"
  >
    <button
      className="styles__IconButton___1LWwV"
      color="primary"
      disabled={true}
      onClick={[Function]}
      size="small"
    >
      T
    </button>
  </div>
  <div
    className="alphabet-letter"
  >
    <button
      className="styles__IconButton___1LWwV"
      color="primary"
      disabled={true}
      onClick={[Function]}
      size="small"
    >
      U
    </button>
  </div>
  <div
    className="alphabet-letter"
  >
    <button
      className="styles__IconButton___1LWwV"
      color="primary"
      disabled={true}
      onClick={[Function]}
      size="small"
    >
      V
    </button>
  </div>
  <div
    className="alphabet-letter"
  >
    <button
      className="styles__IconButton___1LWwV"
      color="primary"
      disabled={true}
      onClick={[Function]}
      size="small"
    >
      W
    </button>
  </div>
  <div
    className="alphabet-letter"
  >
    <button
      className="styles__IconButton___1LWwV"
      color="primary"
      disabled={true}
      onClick={[Function]}
      size="small"
    >
      X
    </button>
  </div>
  <div
    className="alphabet-letter"
  >
    <button
      className="styles__IconButton___1LWwV"
      color="primary"
      disabled={true}
      onClick={[Function]}
      size="small"
    >
      Y
    </button>
  </div>
  <div
    className="alphabet-letter"
  >
    <button
      className="styles__IconButton___1LWwV"
      color="primary"
      disabled={true}
      onClick={[Function]}
      size="small"
    >
      Z
    </button>
  </div>
</div>
`;
